### PR TITLE
[Sema] Fix storage of availability macros definition for SourceKit

### PR DESCRIPTION
**Pull request by xymus**
Copy the availability macros definition from the program command line arguments into LangOptions. This should fix non-deterministic failures at parsing them in SourceKit.

rdar://72356584

### CODE REVIEW

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Comment by xymus** [comment]
@benlangmuir Thanks again for your help with this! And can you tell me if the SourceKit test looks ok?

File: test/SourceKit/Sema/availability_define.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// RUN: %sourcekitd-test -req=sema %s -- %s \
+// RUN:  -Xfrontend -define-availability \
+// RUN:  -Xfrontend "_iOS8Aligned:macOS 10.10, iOS 8.0" | %FileCheck %s
+
+// REQUIRES: OS=macosx
+
+@available(_iOS8Aligned, *)
+public func onMacOS10_10() {}
+// CHECK-NOT: key.line: 7,
```
Comment (benlangmuir) [inline test/SourceKit/Sema/availability_define.swift:9]: Nitpick: can this be changed to `CHECK-NOT: key.line` or maybe `CHECK-NOT: key.description`?  Checking specific line numbers in a "-NOT" check would be really easy to accidentally break without us noticing the test was wrong.

Comment (xymus) [reply test/SourceKit/Sema/availability_define.swift]: Good point, I've updated the test.

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Review by benlangmuir — APPROVED** [review]
Thanks!

**Pull request merged**